### PR TITLE
DOCS-#3936: Update copyright year

### LINKED ISSUE #3936: Update copyright year

### PR DESCRIPTION
**Pull request by YarShev**
Signed-off-by: Igoshev, Yaroslav <yaroslav.igoshev@intel.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #3936  
- [x] tests passing
- [x] module layout described at `docs/developer/architecture.rst` is up-to-date

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -48,7 +48,7 @@ def noop_decorator(*args, **kwargs):
 export_config_help(configs_file_path)
 
 project = "Modin"
-copyright = "2018-2021, Modin"
+copyright = "2018-2022, Modin"
```
Comment (jeffreykennethli) [inline docs/conf.py:51]: is there any reason this is "Modin" and not "Modin Developers" or vice versa?

Comment (YarShev) [reply docs/conf.py:51]: This copyright is shown in our documentation. "Modin Developers" would look strange to me.
image

Comment (devin-petersohn) [reply docs/conf.py:51]: I think it is weird that we have `Modin Developers` in the `NOTICE` file, but not here, we can change it in a new PR.

@jeffreykennethli do you want to open a new issue for it?

**Review by devin-petersohn — APPROVED** [review]
@YarShev LGTM, thanks!

File: docs/conf.py (hunk 1 above)
Comment (jeffreykennethli) [reply docs/conf.py:51]: opened https://github.com/modin-project/modin/issues/3962

**Pull request merged**